src/nupic/algorithms/Cells4.cpp

```diff
@@ -1336,7 +1336,7 @@ void Cells4::compute(Real* input, Real* output, bool doInference, bool doLearnin
   // invalidating our indexes.
   memset(output, 0, _nCells * sizeof(output[0])); // most output is zero
 #if SOME_STATES_NOT_INDEXED
-#ifdef NTA_PLATFORM_darwin86
+#if defined(NTA_ARCH_32) && defined(NTA_OS_DARWIN)
```

Comment:
So the whole block of code below is for 32 bit Darwin systems? Do we really need to account for this?